Source/Fuse.Controls.Video/Video.uno

```diff
@@ -485,6 +487,12 @@ namespace Fuse.Controls
 
 		void OnProgressChanged(object sender, EventArgs args)
 		{
+			lock (this)
+			{
+				_outPosition = Position;
+				_outDuration = Duration;
+			}
```

Comment:
I would kind of prefer if this passed Position and Duration off to a set-method `Video.ScriptClass.uno`, so the state and mutex-locking is contained together in there.

Reply:
Not sure i understand what you mean by this..

Reply:
I mean something like this:
```patch
diff --git a/Source/Fuse.Controls.Video/Video.ScriptClass.uno b/Source/Fuse.Controls.Video/Video.ScriptClass.uno
index ff12d13..0ba31ea 100644
--- a/Source/Fuse.Controls.Video/Video.ScriptClass.uno
+++ b/Source/Fuse.Controls.Video/Video.ScriptClass.uno
@@ -17,6 +17,19 @@ namespace Fuse.Controls
 				new ScriptMethod<Video>("stop", stop, ExecutionThread.MainThread));
 		}
 
+ 		// private fields used by ScriptClass for retrieving the position from JavaScript
+		double _outPosition;
+		double _outDuration;
+		void UpdateScriptClass(double position, double duration)
+		{
+			lock (this)
+			{
+				_outPosition = position;
+				_outDuration = duration;
+			}
+		}
+
+
 		/**
 			Gets the current position in the Video in seconds.
 
diff --git a/Source/Fuse.Controls.Video/Video.uno b/Source/Fuse.Controls.Video/Video.uno
index 09b0786..93379b0 100644
--- a/Source/Fuse.Controls.Video/Video.uno
+++ b/Source/Fuse.Controls.Video/Video.uno
@@ -446,7 +446,6 @@ namespace Fuse.Controls
 			set { if (Playback != null) (Playback as IPlayback).Progress = value; }
 		}
 
-		double _outPosition; //private field used by ScriptClass for retrieving the position from JavaScript
 		/*** The position of the video in seconds */
 		public double Position
 		{
@@ -454,7 +453,6 @@ namespace Fuse.Controls
 			set { if (Playback != null) Playback.Position = value; }
 		}
 
-		double _outDuration; //private field used by ScriptClass for retrieving the duration from JavaScript
 		public double Duration
 		{
 			get { return Playback != null ? Playback.Duration : 0.0; }
@@ -487,12 +485,7 @@ namespace Fuse.Controls
 
 		void OnProgressChanged(object sender, EventArgs args)
 		{
-			lock (this)
-			{
-				_outPosition = Position;
-				_outDuration = Duration;
-			}
-
+			UpdateScriptClass(Position, Duration);
 			OnPropertyChanged(_progressName);
 			if (ProgressChanged != null)
 				ProgressChanged(this, new ValueChangedArgs<double>(Progress));
```